File: px-checkout/src/main/java/com/mercadopago/android/px/internal/mappers/SplitHeaderMapper.java
package com.mercadopago.android.px.internal.mappers;

import androidx.annotation.NonNull;
import com.mercadopago.android.px.internal.features.express.slider.SplitPaymentHeaderAdapter;
import com.mercadopago.android.px.internal.repository.AmountConfigurationRepository;
import com.mercadopago.android.px.model.AmountConfiguration;
import com.mercadopago.android.px.model.Currency;
import com.mercadopago.android.px.model.internal.OneTapItem;

public class SplitHeaderMapper extends Mapper<OneTapItem, SplitPaymentHeaderAdapter.Model> {

    @NonNull private final Currency currency;
    @NonNull private final AmountConfigurationRepository amountConfigurationRepository;

    public SplitHeaderMapper(@NonNull final Currency currency,
        @NonNull final AmountConfigurationRepository amountConfigurationRepository) {
        this.currency = currency;
        this.amountConfigurationRepository = amountConfigurationRepository;
    }

    @Override
    public SplitPaymentHeaderAdapter.Model map(@NonNull final OneTapItem val) {
        if (val.isCard() && val.getStatus().isEnabled()) {
            final String cardId = val.getCard().getId();
            final AmountConfiguration config =
                amountConfigurationRepository.getConfigurationSelectedFor(cardId);
            return config.allowSplit() ? new SplitPaymentHeaderAdapter.SplitModel(currency,
                config.getSplitConfiguration())
                : new SplitPaymentHeaderAdapter.Empty();
        }
        return new SplitPaymentHeaderAdapter.Empty();
    }
}

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/mappers/SplitHeaderMapper.kt
package com.mercadopago.android.px.internal.mappers

import com.mercadopago.android.px.internal.features.express.slider.SplitPaymentHeaderAdapter
import com.mercadopago.android.px.internal.features.express.slider.SplitPaymentHeaderAdapter.SplitModel
import com.mercadopago.android.px.internal.repository.AmountConfigurationRepository
import com.mercadopago.android.px.model.Currency
import com.mercadopago.android.px.model.internal.OneTapItem

internal class SplitHeaderMapper(private val currency: Currency,
    private val amountConfigurationRepository: AmountConfigurationRepository)
    : Mapper<OneTapItem, SplitPaymentHeaderAdapter.Model>() {

    override fun map(value: OneTapItem): SplitPaymentHeaderAdapter.Model {
        return value.takeIf { it.isCard && it.status.isEnabled }?.let { oneTapItem ->
            val cardId = oneTapItem.card.id
            val config = amountConfigurationRepository.getConfigurationSelectedFor(cardId)
            config?.takeIf { it.allowSplit() }?.let {
                val splitConfiguration = config.splitConfiguration
                    ?: throw IllegalStateException("split configuration is mandatory")
                SplitModel(currency, splitConfiguration)
            }
        } ?: SplitPaymentHeaderAdapter.Empty()
    }

}